src/main/scala/viper/server/vsi/HttpVerificationServerInterface.scala

```diff
@@ -0,0 +1,135 @@
+    * */
+  def discardJObConfirmation(jid: Int, msg: String): ToResponseMarshallable
+
+  /** Implement VerificationServer- specific handling of a failed request for for discarding a job
+    * */
+  def discardJobRejection(jid: Int): ToResponseMarshallable
+
+  override def routes(): Route = {
+    /** Send GET request to "/exit".
+      */
+    path("exit") {
+      get {
+        onComplete(getInterruptFutureList()) { err: Try[List[String]] =>
+          err match {
+            case Success(_) =>
+              _termActor ! Terminator.Exit
+              complete( serverStopConfirmation(err) )
+            case Failure(err_msg) =>
+              println(s"Interrupting one of the verification threads timed out: $err_msg")
+              _termActor ! Terminator.Exit
+              complete( serverStopConfirmation(err) )
+          }
+        }
+      }
+    }
+  } ~ path("verify") {
+    /** Send POST request to "/verify".
+      */
+    post {
+      entity(as[Requests.VerificationRequest]) { r =>
+        complete(onVerifyPost(r))
+      }
+    }
+  } ~ path("verify" / IntNumber) { jid =>
+    /** Send GET request to "/verify/<jid>" where <jid> is a non-negative integer.
+      * <jid> must be an ID of an existing verification job.
+      */
+    get {
+      jobs.lookupJob(jid) match {
+        case Some(handle_future) =>
+          // Found a job with this jid.
+          onComplete(handle_future) {
+            case Success(handle) =>
+              val s: Source[Envelope, NotUsed] = Source.fromPublisher((handle.publisher))
+              _termActor ! Terminator.WatchJobQueue(jid, handle)
+              complete(unpackMessages(s))
+            case Failure(error) =>
+              complete(verificationRequestRejection(jid, error))
+          }
+        case _ =>
+          complete(verificationRequestRejection(jid, JobNotFoundException()))
+      }
+    }
+  } ~ path("discard" / IntNumber) { jid =>
+    /** Send GET request to "/discard/<jid>" where <jid> is a non-negative integer.
+      * <jid> must be an ID of an existing verification job.
+      */
+    get {
+      jobs.lookupJob(jid) match {
+        case Some(handle_future) =>
+          onComplete(handle_future) {
+            case Success(handle) =>
+              implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
+              val interrupt_done: Future[String] = (handle.controller_actor ? Stop(true)).mapTo[String]
+              onSuccess(interrupt_done) { msg =>
+                handle.controller_actor ! PoisonPill // the actor played its part.
+                complete(discardJObConfirmation(jid, msg))
+              }
+            case Failure(_) =>
+              complete(discardJobRejection(jid))
+          }
+        case _ =>
+          complete(discardJobRejection(jid))
+      }
+    }
+  }
+}
+
+object Requests extends akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport with DefaultJsonProtocol {
```

Comment:
This is part of what's called a protocol in AkkaHttp. Please put it into a separate file.

Reply:
There are several protocol objects within VSI. Can I aggregate them into a Protocol.scala file, or should I have one file per type of protocol?